Documentation/tutorial/howto_create_custom_plugin.md

```diff
@@ -0,0 +1,160 @@
+How-to: Create Build Custom Document
+====================================
+
+In this topic, we will create a plug-in to build some simple rtf files to documents.
+
+Agenda
+------
+* [Goal and limitation](#goal-and-limitation)
+* [Prepare](#prepare)
+* [Create a document processor](#create-a-document-processor)
+* [Create a document build step](#create-a-document-build-step)
+* [Enable plug-in](#enable-plug-in)
+* [Build document](#build-document)
+
+Goal and limitation
+-------------------
+1.  In scope:
+    1.  Our input will be a set of rtf files, with `.rtf` extension file name.
+    2.  The rtf files will be build as conceptual document.
+2.  Out of scope:
+    1.  No picture or other object in rtf files.
+    2.  No hyper link in rtf files.
+    3.  No metadata.
+
+Prepare
+-------
+1.  Create a new library project in `Visual Studio`.
+
+2.  Add nuget packages:  
+    * `System.Collections.Immutable` with version 1.1.37
+    * `System.Composition`with version 1.0.27
+
+3.  Add nuget package `Microsoft.DocAsCode.Plugins` same version with DocFX, or reference the project.
+
+4.  Add framework assembly reference:
+    `PresentationCore`, `PresentationFramework`, `WindowsBase`
+
+5.  Include some code files:
+    rtf to html converter: [MarkupConverter](https://github.com/mmanela/MarkupConverter)
```

Comment:
are there any nuget packages for the converter? Referencing packages seems more straight forward